core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorUtil.java

```diff
@@ -1375,6 +1376,14 @@ private static void adjustTypeForMultisetConstructor(
       } else {
         elementType = oddType;
       }
+      if (call.getOperator().getName().equals("ARRAY_INSERT") && i == 1) {
```

Comment:
pls don't modify `adjustTypeForMultisetConstructor` directly. just add a static method to cast inserted element to array componentType.

Reply:
do not hard-code here.

Reply:
done